cob_scan_unifier/src/scan_unifier_node.cpp

```diff
@@ -58,30 +58,79 @@
 
 #include <cob_scan_unifier/scan_unifier_node.h>
 
-// constructor
-scan_unifier_node::scan_unifier_node()
+// Constructor
+ScanUnifierNode::ScanUnifierNode()
 {
-  ROS_DEBUG("init laser unification");
-  // create node handles
+  ROS_DEBUG("Init scan_unifier");
+
+  // Create node handles
   nh_ = ros::NodeHandle();
   pnh_ = ros::NodeHandle("~");
 
-  // publisher
+  // Publisher
   topicPub_LaserUnified_ = nh_.advertise<sensor_msgs::LaserScan>("scan_unified", 1);
 
   getParams();
 
-  ROS_DEBUG("scan unifier: now init laser structs");
+  // Subscribe to Laserscan topics
+
+  for(int i = 0; i < config_.number_input_scans; i++)
+  {
+    message_filter_subscribers_.push_back(new message_filters::Subscriber<sensor_msgs::LaserScan>(nh_, config_.input_scan_topics.at(i), 1));
+  }
+
+
+  // Initialize message_filters::Synchronizer with the right constructor for the choosen number of inputs.
 
-  pthread_mutex_init(&m_mutex, NULL);
+  switch (config_.number_input_scans)
+  {
+    case 2:
+    {
+      typedef message_filters::sync_policies::ApproximateTime<sensor_msgs::LaserScan, sensor_msgs::LaserScan> SyncPolicy;
+      synchronizer2_ = new message_filters::Synchronizer<SyncPolicy>(SyncPolicy(2), *message_filter_subscribers_.at(0), *message_filter_subscribers_.at(1));
+      synchronizer2_->setInterMessageLowerBound(0, ros::Duration(0.167));
+      synchronizer2_->setInterMessageLowerBound(1, ros::Duration(0.167));
```

Comment:
Well, I meant **after** successful testing we could at least introduce parameters to make the scan_unifier node configurable...but those parameters would be scan_unifier parameters and not related to parameters (of the same meaning) of the scanners, though